generator/src/main/java/com/scottlogic/deg/generator/outputs/targets/ViolationDirectoryOutputTarget.java

```diff
@@ -0,0 +1,54 @@
+package com.scottlogic.deg.generator.outputs.targets;
+
+import com.scottlogic.deg.generator.Profile;
+import com.scottlogic.deg.generator.outputs.formats.OutputFormat;
+import com.scottlogic.deg.generator.utils.FileUtils;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+/** Represents a directory specified by a user as a target for violation data */
+public class ViolationDirectoryOutputTarget implements MultiDatasetOutputTarget {
+    private final FileUtils fileUtils;
+    private final Path directoryPath;
+    private final boolean canOverwriteExistingFiles;
+    private final OutputFormat formatOfViolationDatasets;
+
+    public ViolationDirectoryOutputTarget(
+        Path directoryPath,
+        OutputFormat formatOfViolationDatasets,
+        boolean canOverwriteExistingFiles,
+        FileUtils fileUtils) {
+
+        this.fileUtils = fileUtils;
+        this.directoryPath = directoryPath;
+        this.canOverwriteExistingFiles = canOverwriteExistingFiles;
+        this.formatOfViolationDatasets = formatOfViolationDatasets;
+    }
+
+    @Override
+    public SingleDatasetOutputTarget getSubTarget(String name) {
+        String filename =
+            formatOfViolationDatasets.getFileExtensionWithoutDot()
+                .map(extension -> name + "." + extension)
+                .orElse(name);
+
+        return new FileOutputTarget(
+            directoryPath.resolve(filename),
+            formatOfViolationDatasets, canOverwriteExistingFiles, fileUtils
+        );
+    }
+
+    @Override
+    public void validate(Profile profile) throws OutputTargetValidationException, IOException {
+        if (!fileUtils.exists(directoryPath)) {
+            fileUtils.createDirectories(directoryPath);
+        } else if (!fileUtils.isDirectory(directoryPath)) {
+            throw new OutputTargetValidationException(
+                "not a directory, please enter a valid directory name");
+        } else if (!canOverwriteExistingFiles && !fileUtils.isDirectoryEmpty(directoryPath, profile.rules.size())) {
+            throw new OutputTargetValidationException(
+                "directory not empty, please remove any 'manifest.json' and '[0-9].csv' files or use the --replace option");
```

Comment:
I'm not sure I like this error message. It's confusing because the directory doesn't have to be empty - it just mustn't contain files that might be overwritten.  My response to it would be "but if I remove those files then the directory still won't be empty".

It's also slightly confusing because if a profile contains, say, 37 rules, then strictly following the instruction to remove files [0-9].csv still won't resolve the situation.

Reply:
Agreed! (but that message existed before this PR, so it might be better to treat fixing it as a separate bit of work - especially since some of the solutions would change the behaviour and not just the validation message?)